couch2pg/src/importer.js

```diff
@@ -0,0 +1,132 @@
+const BATCH_SIZE = process.env.BATCH_SIZE || 1000;
+
+import * as db from './db.js';
+
+const SELECT_SEQ_STMT = `SELECT seq FROM ${db.postgresProgressTable} WHERE source = $1`;
+const INSERT_SEQ_STMT = `INSERT INTO ${db.postgresProgressTable}(seq, source) VALUES ($1, $2)`;
+const UPDATE_SEQ_STMT = `UPDATE ${db.postgresProgressTable} SET seq = $1 WHERE source = $2`;
+const INSERT_DOCS_STMT = `INSERT INTO ${db.postgresTable} ("@timestamp", _id, _rev, doc) VALUES`;
+const ON_CONFLICT_STMT = 'ON CONFLICT (_id, _rev) DO NOTHING';
```

Comment:
Is it possible to have a conflict on both the _id and _rev values? I assumed the conflict would only be on the _id value.

Reply:
Yes, i think this can happen if the container fails before we successfully update the seq. also Couchdb replication protocol specifically mentions that you can get the same change twice.